src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/transport/action/updateindexmetadata/TransportUpdateManagedIndexMetaDataAction.kt

```diff
@@ -54,7 +54,10 @@ class TransportUpdateManagedIndexMetaDataAction : TransportMasterNodeAction<Upda
     )
 
     override fun checkBlock(request: UpdateManagedIndexMetaDataRequest, state: ClusterState): ClusterBlockException? {
-        return state.blocks.globalBlockedException(ClusterBlockLevel.METADATA_WRITE)
+        // https://github.com/elastic/elasticsearch/commit/ae14b4e6f96b554ca8f4aaf4039b468f52df0123
+        // This commit will help us to give each individual index name and the error that is cause it. For now it will be a generic error message.
+        val indices = request.listOfIndexMetadata.map { it.first.name }.toTypedArray()
+        return state.blocks.indicesBlockedException(ClusterBlockLevel.METADATA_WRITE, indices)
```

Comment:
Do we still want to check for the globalBlock or is it automatically covered by checking for indicesBlock?

Reply:
`indicesBlockedException` checks for `globalBlock` also.